glommio/src/io/read_result.rs

```diff
@@ -23,7 +23,7 @@ impl core::ops::Deref for ReadResult {
 #[derive(Clone, Debug)]
 struct ReadResultInner {
     buffer: ScheduledSource,
-    offset: usize,
+    mem: *const u8,
```

Comment:
can we at least use `NonNull` here?

If this is null your unsafe `deref` crashes.